backend/seafile/seafile_internal_test.go

```diff
@@ -0,0 +1,187 @@
+	expectedDirectory := "path"
+
+	fs := &Fs{
+		libraryName:   expectedLibrary,
+		rootDirectory: "",
+	}
+	libraryName, directory := fs.splitPath(expectedDirectory)
+	if libraryName != expectedLibrary {
+		t.Errorf("library name should be '%s' but found '%s'", expectedLibrary, libraryName)
+	}
+	if directory != expectedDirectory {
+		t.Errorf("directory should be '%s' but found '%s'", expectedDirectory, directory)
+	}
+}
+
+func TestSubfoldersSplitPathFromRootLibraryTarget(t *testing.T) {
+	expectedLibrary := "Library"
+	expectedDirectory := path.Join("path", "to", "file")
+
+	fs := &Fs{
+		libraryName:   expectedLibrary,
+		rootDirectory: "",
+	}
+	libraryName, directory := fs.splitPath(expectedDirectory)
+	if libraryName != expectedLibrary {
+		t.Errorf("library name should be '%s' but found '%s'", expectedLibrary, libraryName)
+	}
+	if directory != expectedDirectory {
+		t.Errorf("directory should be '%s' but found '%s'", expectedDirectory, directory)
+	}
+}
+
+func TestEmptySplitPathFromFullPathTarget(t *testing.T) {
+	expectedLibrary := "Library"
+	expectedRoot := "root"
+
+	fs := &Fs{
+		libraryName:   expectedLibrary,
+		rootDirectory: expectedRoot,
+	}
+	libraryName, directory := fs.splitPath("")
+	if libraryName != expectedLibrary {
+		t.Errorf("library name should be '%s' but found '%s'", expectedLibrary, libraryName)
+	}
+	if directory != expectedRoot {
+		t.Errorf("directory should be '%s' but found '%s'", expectedRoot, directory)
+	}
+}
+
+func TestSingleFolderSplitPathFromFullPathTarget(t *testing.T) {
+	expectedLibrary := "Library"
+	expectedRoot := "root"
+	subPath := "path"
+	expectedDirectory := path.Join(expectedRoot, subPath)
+
+	fs := &Fs{
+		libraryName:   expectedLibrary,
+		rootDirectory: expectedRoot,
+	}
+	libraryName, directory := fs.splitPath(subPath)
+	if libraryName != expectedLibrary {
+		t.Errorf("library name should be '%s' but found '%s'", expectedLibrary, libraryName)
+	}
+	if directory != expectedDirectory {
+		t.Errorf("directory should be '%s' but found '%s'", expectedDirectory, directory)
+	}
+}
+
+func TestSubfoldersSplitPathFromFullPathTarget(t *testing.T) {
+	expectedLibrary := "Library"
+	expectedRoot := "root"
+	subPath := path.Join("path", "to", "file")
+	expectedDirectory := path.Join(expectedRoot, subPath)
+
+	fs := &Fs{
+		libraryName:   expectedLibrary,
+		rootDirectory: expectedRoot,
+	}
+	libraryName, directory := fs.splitPath(subPath)
+	if libraryName != expectedLibrary {
+		t.Errorf("library name should be '%s' but found '%s'", expectedLibrary, libraryName)
+	}
+	if directory != expectedDirectory {
+		t.Errorf("directory should be '%s' but found '%s'", expectedDirectory, directory)
+	}
+}
+
+func TestSplitPathIntoSlice(t *testing.T) {
+	testData := map[string][]string{
+		"1":     {"1"},
+		"/1":    {"1"},
+		"/1/":   {"1"},
+		"1/2/3": {"1", "2", "3"},
+	}
+	for input, expected := range testData {
+		output := splitPath(input)
+		if !equalSlice(output, expected) {
```

Comment:
You can just use `assert.Equal(t, expected, output)` (in that order!) here so the `equalSlice` function isn't necessary.